data/json/mapgen/homeimprovement_superstore.json

```diff
@@ -0,0 +1,291 @@
+[
+    {
+        "type"  : "item_group",
+        "id"    : "pwr_clothing",
+        "items" : [
+            ["hat_hard",         100],
+            ["boots",            35],
+            ["glasses_safety",   50]
+        ]
+    },
+        {
+        "type"  : "item_group",
+        "id"    : "pwr_tools",
+        "items" : [
+            ["flashlight",       35],
+            ["wrench",           50],
+            ["screwdriver",      35]
+        ]
+    },
+        {
+        "type"  : "item_group",
+        "id"    : "pwr_parts",
+        "items" : [
+            ["cable",            65],
+            ["power_supply",     50],
+            ["amplifier",        50]
+        ]
+    },
+	    {
+        "type"  : "item_group",
+        "id"    : "wood",
+        "items" : [
+            ["2x4",            80],
+            ["power_supply",     50],
+            ["amplifier",        50]
+        ]
+    },
+    {
+        "type" : "mapgen",
+        "om_terrain" : ["hdwr_large_entrance"],
+        "method": "json",
+        "weight": 100,
+        "object": {
+            "rows":[
+            ".|s....................|",
+            ".|t.............rrrrrrr|",
+            ".......................|",
+            ".................ccc...|",
+            ".---.....r|r...........|",
+            ".|CT.....r|r.....ccc...|", 
```

Comment:
Good to know, re the bug.

However, I think you're on the wrong track. Not having a terrain mapping for the letter C might be an error, but it's not the error that throws this specific exception.

                pjo = jo.get_object("terrain");
                for( const auto &key : pjo.get_member_names() ) {
                    if( pjo.has_string( key ) ) {
                        const auto tmpval = pjo.get_string( key );
                        const auto iter = termap.find( tmpval );
                        if( iter == termap.end() ) {
                            jo.throw_error( "Invalid terrain", key );
                        }

it looks like the "Invalid terrain" error specifically means that there IS a defined terrain mapping for a letter, but that the mapped string doesn't exist in termap[]